File: connectx-backend/src/main/java/com/connectx/device/dto/RegisterDeviceDto.java
package com.connectx.device.dto;

import jakarta.validation.constraints.NotBlank;

public class RegisterDeviceDto {

    @NotBlank(message = "Device name is required")
    private String deviceName;

    @NotBlank(message = "Public key is required")
    private String publicKey;

    @NotBlank(message = "Key algorithm is required")
    private String keyAlgorithm; // e.g. "ECDH-P256", "RSA-OAEP"

    public RegisterDeviceDto() {}

    public String getDeviceName() {
        return deviceName;
    }

    public void setDeviceName(String deviceName) {
        this.deviceName = deviceName;
    }

    public String getPublicKey() {
        return publicKey;
    }

    public void setPublicKey(String publicKey) {
        this.publicKey = publicKey;
    }

    public String getKeyAlgorithm() {
        return keyAlgorithm;
    }

    public void setKeyAlgorithm(String keyAlgorithm) {
        this.keyAlgorithm = keyAlgorithm;
    }
}

File: connectx-android/app/src/main/java/com/connectx/app/data/remote/device/model/RegisterDeviceDto.kt
package com.connectx.app.data.remote.device.model

import kotlinx.serialization.Serializable

/**
 * Mirrors com.connectx.device.dto.RegisterDeviceDto exactly. All three fields are
 * backend @NotBlank -- required, non-nullable here. `publicKey`/`keyAlgorithm` are
 * opaque E2EE metadata (e.g. "ECDH-P256") -- no crypto behavior implemented here,
 * per N7's boundary.
 */
@Serializable
data class RegisterDeviceDto(
    val deviceName: String,
    val publicKey: String,
    val keyAlgorithm: String
)
